perception/autoware_tensorrt_bevdet/src/bevdet_node.cpp

```diff
@@ -0,0 +1,342 @@
+
+  startCameraInfoSubscription();
+
+  // Wait for camera info and tf transform initialization
+  timer_ = this->create_wall_timer(
+    std::chrono::milliseconds(100), std::bind(&TRTBEVDetNode::checkInitialization, this));
+}
+
+void TRTBEVDetNode::initModel()
+{
+  score_thre_ = this->declare_parameter<float>("post_process_params.score_threshold", 0.2);
+
+  model_config_ = this->declare_parameter("model_config", "bevdet_r50_4dlongterm_depth.yaml");
+
+  onnx_file_ = this->declare_parameter<std::string>("onnx_path", "bevdet_one_lt_d.onnx");
+  engine_file_ = this->declare_parameter<std::string>("engine_path", "bevdet_one_lt_d.engine");
+
+  imgs_name_ = this->declare_parameter<std::vector<std::string>>("data_params.cams");
+  class_names_ =
+    this->declare_parameter<std::vector<std::string>>("post_process_params.class_names");
+
+  RCLCPP_INFO_STREAM(this->get_logger(), "Successful load config!");
+
+  sampleData_.param = camParams(cams_intrin, cams2ego_rot, cams2ego_trans);
+
+  RCLCPP_INFO_STREAM(this->get_logger(), "Successful load image params!");
+
+  bevdet_ = std::make_shared<autoware::tensorrt_bevdet::BEVDet>(
+    model_config_, img_N_, sampleData_.param.cams_intrin, sampleData_.param.cams2ego_rot,
+    sampleData_.param.cams2ego_trans, onnx_file_, engine_file_);
+
+  RCLCPP_INFO_STREAM(this->get_logger(), "Successful create bevdet!");
+
+  CHECK_CUDA(cudaMalloc(
+    reinterpret_cast<void **>(&imgs_dev_), img_N_ * 3 * img_w_ * img_h_ * sizeof(uchar)));
+
+  pub_boxes_ = this->create_publisher<autoware_perception_msgs::msg::DetectedObjects>(
+    "~/output/boxes", rclcpp::QoS{1});
+}
+
+void TRTBEVDetNode::checkInitialization()
+{
+  if (camera_info_received_flag_) {
+    RCLCPP_INFO_STREAM(
+      this->get_logger(), "Camera Info and TF Transform Initialization completed!");
+    initModel();
+    startImageSubscription();
+    timer_->cancel();
+    timer_.reset();
+  } else {
+    RCLCPP_INFO_THROTTLE(
+      this->get_logger(), *this->get_clock(), 5000,
+      "Waiting for Camera Info and TF Transform Initialization...");
+  }
+}
+
+void TRTBEVDetNode::startImageSubscription()
+{
+  using std::placeholders::_1;
+  using std::placeholders::_2;
+  using std::placeholders::_3;
+  using std::placeholders::_4;
+  using std::placeholders::_5;
+  using std::placeholders::_6;
+
+  sub_f_img_.subscribe(
+    this, "~/input/topic_img_f",
+    rclcpp::QoS{1}.get_rmw_qos_profile());  // rmw_qos_profile_sensor_data
+  sub_b_img_.subscribe(this, "~/input/topic_img_b", rclcpp::QoS{1}.get_rmw_qos_profile());
+
+  sub_fl_img_.subscribe(this, "~/input/topic_img_fl", rclcpp::QoS{1}.get_rmw_qos_profile());
+  sub_fr_img_.subscribe(this, "~/input/topic_img_fr", rclcpp::QoS{1}.get_rmw_qos_profile());
+
+  sub_bl_img_.subscribe(this, "~/input/topic_img_bl", rclcpp::QoS{1}.get_rmw_qos_profile());
+  sub_br_img_.subscribe(this, "~/input/topic_img_br", rclcpp::QoS{1}.get_rmw_qos_profile());
+
+  sync_ = std::make_shared<Sync>(
```

Comment:
The `Sync` object uses a message queue with a size of 10. While this might work in most situations, a larger queue might be necessary to handle temporary message drops or bursts, especially if the camera stream is irregular or experiences delays. Consider increasing this queue size to 100 or configuring it dynamically based on system parameters.  Otherwise, message loss can lead to inconsistencies and inaccurate detection.